src/chocolatey/infrastructure.app/commands/ChocolateyCommandBase.cs

```diff
@@ -13,16 +13,177 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-using System.Collections.Generic;
-using System.Linq;
-
 namespace chocolatey.infrastructure.app.commands
-{
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text;
+    using chocolatey.infrastructure.app.attributes;
+    using chocolatey.infrastructure.app.configuration;
+    using chocolatey.infrastructure.commands;
+    using chocolatey.infrastructure.logging;
+
     /// <summary>
     /// A base class for any Chocolatey commands which need to utilise shared logic.
     /// </summary>
     public abstract class ChocolateyCommandBase
-    {
+    {
+        public virtual void HelpMessage(ChocolateyConfiguration configuration)
+        {
+            var allCommandForAttributes = GetType().GetCustomAttributes<CommandForAttribute>().ToArray();
+            var commandForAttribute = allCommandForAttributes.OrEmpty().FirstOrDefault();
+
+            if (commandForAttribute == null)
+            {
+                throw new ApplicationException("No command for attribute was found on the command class: {0}".FormatWith(GetType().Name));
+            }
+
+            this.Log().Info(ChocolateyLoggers.Important, "{0} Command", GetCommandName(commandForAttribute));
+
+            if (!string.IsNullOrEmpty(commandForAttribute.Version))
+            {
+                this.Log().Info(@"
+WARNING: {0}", GetCommandVersionWarning(commandForAttribute));

```

Comment:
I see that the WARNING is still on this line. Are you thinking that this is a requirement?

Reply:
This has been answered privately, but I will repeat it here.

Yes, this was intentional to have added. The idea is that the `WARNING` will flow through to the Documentation website to have this warning show up there.

Reply:
This will show up on the web page with my last commit like this:

![image](https://github.com/chocolatey/choco/assets/1474648/deef4a60-cd34-448d-bc88-6f19d767f42e)